Feature/replies

sherryliao21 opened the pull request:
Please review my code, which has passed all unit tests and POSTMAN test case, thanks!

sherryliao21 committed:
feat: add route POST /:tweet_id/replies

sherryliao21 committed:
feat: add route GET /api/tweets/:tweet_id/replies

sherryliao21 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [GET /api/tweets/:tweet_id/replies](https://trello.com/c/XwiGHa9S/67-get-api-tweets-tweetid-replies)

sherryliao21 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [POST /api/tweets/:tweet_id/replies](https://trello.com/c/Kco4t7gW/66-post-api-tweets-tweetid-replies)

ivyhungtw commented on controllers/tweetController.js:
```diff
@@ -113,6 +113,73 @@ const tweetController = {
+
+      const tweetAuthor = await User.findOne({ where: { id: targetTweet.UserId } })
+
+      if (!req.body.comment || req.body.comment === '') {
```
empty string 是 falsy value，所以 req.body.comment === '' 不用加也可以

sherryliao21 replied:
已修改為 `if (!req.body.comment.trim()) {...}`

ivyhungtw commented on controllers/tweetController.js:
```diff
@@ -113,6 +113,73 @@ const tweetController = {
+      const replies = await Reply.findAll({ raw: true, nest: true, where: { TweetId } })
+      return res.json(
+        replies
+        // 因測試檔希望回傳的 res.body 是陣列，並且此陣列[0] 就是第一個 reply data，故只能將 res.json() 中的大花括號拿掉以符合測試格式。POSTMAN 測試沒問題。
```
這個沒問題，上次有討論過可以直接回傳 array

sherryliao21 replied:
ok 已移除註解與 status, message，僅回傳 replies data

ivyhungtw commented on routes/modules/tweets.js:
```diff
@@ -55,5 +55,7 @@ router.post('/', authenticated, tweetController.postTweets)
 router.post('/:tweet_id/unlike', authenticated, tweetController.unlikeTweet)
+router.post('/:tweet_id/replies', authenticated, tweetController.postReply)
+router.get('/:tweet_id/replies', authenticated, tweetController.getReplies)
 
```
authenticated 可以拿掉，前面 index.js 有加了

sherryliao21 replied:
ok 已拿掉，也實測能夠正常運作

sherryliao21 committed:
feat: modify tweetController.js && routes/tweets.js
1. optimize controller functions by adding database checks && input checks
2. remove authenticated middleware in routes/tweets.js

sherryliao21 commented:
已將上述問題修正完畢更新在[這支 commit](https://github.com/ivyhungtw/twitter-api-2020/commit/e1fea8017c8b18c94c6668af216ef6695188192e) 中
再麻煩 @ivyhungtw 幫忙 review && approve，感謝

sherryliao21 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [fix /tweets](https://trello.com/c/Y2Ni5Nrg/75-fix-tweets)

sherryliao21 commented:
> 這個的意思是超過 140 自仍然可以發文，只是會被系統自動截斷成 140 字嗎？
對，稍早跟前端確認過他們會做欄位字數驗證來擋，那我就改回原本寫法了，直接傳送錯誤訊息

sherryliao21 committed:
feat: modify tweetController.postTweets
change response: if description > 140 characters, response with error message instead of the content
